.evergreen.yml

```diff
@@ -218,115 +226,173 @@ tasks:
+    commands:
+      - func: checkout
+      - func: install
+      - func: test
+        vars:
+          mongosh_server_test_version: "4.0.x"
+  - name: test_m42x
+    commands:
+      - func: checkout
+      - func: install
+      - func: test
+        vars:
+          mongosh_server_test_version: "4.2.x"
+  - name: test_m44x
     commands:
       - func: checkout
       - func: install
       - func: test
-  - name: test_ps
+        vars:
+          mongosh_server_test_version: "4.4.x"
+  - name: test_m40x_ps
     commands:
       - func: checkout
       - func: install_ps
       - func: test_ps
-  - name: test_vscode
+        vars:
+          mongosh_server_test_version: "4.0.x"
+  - name: test_m42x_ps
     commands:
       - func: checkout
-      - func: install
-      - func: test_vscode
-  - name: release_macos
-    depends_on:
-      - name: check
-      - name: test
+      - func: install_ps
+      - func: test_ps
+        vars:
+          mongosh_server_test_version: "4.2.x"
+  - name: test_m44x_ps
     commands:
       - func: checkout
-      - func: install
-      - func: release_macos
-  - name: release_linux
-    depends_on:
-      - name: check
-      - name: test
+      - func: install_ps
+      - func: test_ps
+        vars:
+          mongosh_server_test_version: "4.4.x"
+  - name: test_vscode
     commands:
       - func: checkout
       - func: install
-      - func: release_linux
-  - name: release_debian
+      - func: test_vscode
+  - name: compile_and_upload_artifact_macos
     depends_on:
       - name: check
-      - name: test
-        variant: linux
+        variant: darwin
+      - name: test_m40x
+        variant: darwin
+      - name: test_m42x
+        variant: darwin
+      - name: test_m44x
+        variant: darwin
     commands:
       - func: checkout
       - func: install
-      - func: release_debian
-  - name: release_rhel
+      - func: compile_and_upload_artifact
+  - name: compile_and_upload_artifact_linux
     depends_on:
       - name: check
-      - name: test
+      - name: test_m40x
+        variant: linux
+      - name: test_m42x
+        variant: linux
+      - name: test_m44x
         variant: linux
     commands:
       - func: checkout
       - func: install
-      - func: release_rhel
-  - name: release_win_ps
+      - func: compile_and_upload_artifact
+  - name: compile_and_upload_artifact_ps
     depends_on:
       - name: check_ps
+      - name: test_m40x_ps
+      - name: test_m42x_ps
+      - name: test_m44x_ps
     commands:
       - func: checkout
       - func: install_ps
-      - func: release_win_ps
+      - func: compile_and_upload_artifact_ps
+  - name: test_linux_artifact
+    depends_on:
+      - name: compile_and_upload_artifact_linux
+        variant: linux
+    commands:
+      - func: checkout
+      - func: install
+      - func: test_linux_artifact
   - name: release_publish
     depends_on:
-      - name: release_win_ps
+      - name: compile_and_upload_artifact_ps
```

Comment:
Probably at this point we can rename `ps` to `win` and `powershell` to `windows`?
I think the only reason we had `ps` was cause at some point we were transitioning from `cygwin` which was the default windows shell in evergreen

Reply:
Sure, done :+1: